rasterio/rio/features.py

```diff
@@ -20,6 +19,128 @@
 logger = logging.getLogger('rio')
 
 
+# Extract command
+@cli.command(short_help='Extract raster using features.')
+@files_inout_arg
+@format_opt
+@click.option('--all_touched', is_flag=True, default=False,
```

Comment:
Yes, let's change rasterize, too. Classic unix style options with dashes as word separators is our style.